src/assistant/SuperflexAssistant.ts

```diff
@@ -182,6 +184,12 @@ export default class SuperflexAssistant implements Assistant {
       )
     }
 
+    if (!(await isUploadAllowedByTierAndSize(documentPaths))) {
+      throw new Error(
+        'Repository is too large to be fully indexed. Upgrade your subscription plan to get results better aligned with your existing codebase.'
```

Comment:
Improve messaging maybe something like
```suggestion
        'This repo is too large to fully index on the free plan. Upgrade to index the entire codebase and get more accurate results.'
```